src/Features/CSharp/Portable/OmitDefaultAccessibilityModifiers/CSharpOmitDefaultAccessibilityModifiersDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+        {
+        }
+
+        protected override void ProcessCompilationUnit(
+            SyntaxTreeAnalysisContext context, SyntaxGenerator generator, 
+            CodeStyleOption<bool> option, CompilationUnitSyntax compilationUnit)
+        {
+            ProcessMembers(context, generator, option, compilationUnit.Members);
+        }
+
+        private void ProcessMembers(
+            SyntaxTreeAnalysisContext context, SyntaxGenerator generator, 
+            CodeStyleOption<bool> option, 
+            SyntaxList<MemberDeclarationSyntax> members)
+        {
+            foreach (var memberDeclaration in members)
+            {
+                ProcessMemberDeclaration(context, generator, option, memberDeclaration);
+            }
+        }
+
+        private void ProcessMemberDeclaration(
+            SyntaxTreeAnalysisContext context, SyntaxGenerator generator,
+            CodeStyleOption<bool> option, MemberDeclarationSyntax member)
+        {
+            if (member.IsKind(SyntaxKind.NamespaceDeclaration, out NamespaceDeclarationSyntax namespaceDeclaration))
+            {
+                ProcessMembers(context, generator, option, namespaceDeclaration.Members);
+            }
+
+            // If we have a class or struct, recurse inwards.
+            if (member.IsKind(SyntaxKind.ClassDeclaration, out TypeDeclarationSyntax typeDeclaration) ||
+                member.IsKind(SyntaxKind.StructDeclaration, out typeDeclaration))
+            {
+                ProcessMembers(context, generator, option, typeDeclaration.Members);
+            }
+
+#if false
+            // Add this once we have the language version for C# that supports accessibility
+            // modifiers on interface methods.
+            if (option.Value == AccessibilityModifiersRequired.Always &&
+                member.IsKind(SyntaxKind.InterfaceDeclaration, out typeDeclaration))
+            {
+                // Only recurse into an interface if the user wants accessibility modifiers on 
+                ProcessTypeDeclaration(context, generator, option, typeDeclaration);
+            }
+#endif
+
+            // Have to have a name to report the issue on.
+            var name = member.GetNameToken();
+            if (name.Kind() == SyntaxKind.None)
+            {
+                return;
+            }
+
+            // Certain members never have accessibility. Don't bother reporting on them.
+            if (!generator.CanHaveAccessibility(member))
+            {
+                return;
+            }
+
+            // If they already don't have an accessibility, no need to report anything.
+            var accessibility = generator.GetAccessibility(member);
+            if (accessibility == Accessibility.NotApplicable)
+            {
+                return;
+            }
+
+            // Check for default modifiers
+            var parentKind = member.Parent?.Kind();
+            if(parentKind == null || parentKind == SyntaxKind.NamespaceDeclaration)
```

Comment:
space after if.

Reply:
i don't see how you could ever have a null parent.

Reply:
What is the parent of a type that's not in a namespace? I can add a test for that if that'd be useful.

Reply:
A CompilationUnitSyntax.  Test would be valuable :)

Reply:
Done. Changed the analyzer and added a test to cover that and it passes. Still stuck on the extra CR being added on the main test. Nothing I've added should be adding a space, so is there an interfering default somewhere?